File: Practice/12/Python/example12.py

stop = 1
while stop == 1:
	stop = 0
	n = int(input())
	resultat = 1
	i = 2
	while i <= n:
		resultat = resultat * i
		i +=1

	if resultat >= 1 and resultat <= 10**9:
		print(resultat)
	else:
		print("Факториал не входит в диапазон")
		stop = 1

File: Practice/12/C++/example12/example12/example12.cpp
#include <iostream>
#include <cmath>
using namespace std;

int main()
{
    setlocale(LC_ALL, "Russian");

    int n, resultat;
    int stop = 1;
    for (; stop == 1;) {
        stop = 0;
        cin >> n;
        resultat = 1;

        for (int i = 2; i <= n; i++)
        {
            resultat = resultat * i;

        }


        if (resultat <= pow(10, 9) and resultat >= 1) {
            cout << resultat;

        }
        else {
            cout << "Факториал не входит в диапазон" << endl;
            stop = 1;

        }
    }
}
